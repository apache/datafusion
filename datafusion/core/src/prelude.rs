// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.pub},

//! A "prelude" for users of the datafusion crate.
//!
//! Like the standard library's prelude, this module simplifies importing of
//! common items. Unlike the standard prelude, the contents of this module must
//! be imported manually:
//!
//! ```
//! use datafusion::prelude::*;
//! ```

pub use crate::dataframe::DataFrame;
pub use crate::execution::context::{SessionConfig, SessionContext};
pub use crate::execution::options::{
    AvroReadOptions, CsvReadOptions, NdJsonReadOptions, ParquetReadOptions,
};

pub use datafusion_common::Column;
pub use datafusion_expr::{
    expr_fn::*,
    lit, lit_timestamp_nano,
    logical_plan::{JoinType, Partitioning},
    Expr,
};
