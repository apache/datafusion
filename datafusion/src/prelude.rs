// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.pub},

//! A "prelude" for users of the datafusion crate.
//!
//! Like the standard library's prelude, this module simplifies importing of
//! common items. Unlike the standard prelude, the contents of this module must
//! be imported manually:
//!
//! ```
//! use datafusion::prelude::*;
//! ```

pub use crate::dataframe::DataFrame;
pub use crate::execution::context::{ExecutionConfig, ExecutionContext};
pub use crate::execution::options::AvroReadOptions;
pub use crate::execution::options::{CsvReadOptions, NdJsonReadOptions};
pub use crate::logical_plan::{
    array, ascii, avg, bit_length, btrim, character_length, chr, col, concat, concat_ws,
    count, create_udf, date_part, date_trunc, digest, in_list, initcap, left, length,
    lit, lower, lpad, ltrim, max, md5, min, now, octet_length, random, regexp_match,
    regexp_replace, repeat, replace, reverse, right, rpad, rtrim, sha224, sha256, sha384,
    sha512, split_part, starts_with, strpos, substr, sum, to_hex, translate, trim, upper,
    Column, JoinType, Partitioning,
};
