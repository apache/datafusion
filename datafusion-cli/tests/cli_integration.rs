// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

use std::process::Command;

use assert_cmd::prelude::{CommandCargoExt, OutputAssertExt};
use predicates::prelude::predicate;
use rstest::rstest;

#[cfg(test)]
#[ctor::ctor]
fn init() {
    // Enable RUST_LOG logging configuration for tests
    let _ = env_logger::try_init();
}

#[rstest]
#[case::exec_from_commands(
    ["--command", "select 1", "--format", "json", "-q"],
    "[{\"Int64(1)\":1}]\n"
)]
#[case::exec_from_files(
    ["--file", "tests/data/sql.txt", "--format", "json", "-q"],
    "[{\"Int64(1)\":1}]\n"
)]
#[case::set_batch_size(
    ["--command", "show datafusion.execution.batch_size", "--format", "json", "-q", "-b", "1"],
    "[{\"name\":\"datafusion.execution.batch_size\",\"setting\":\"1\"}]\n"
)]
#[test]
fn cli_quick_test<'a>(
    #[case] args: impl IntoIterator<Item = &'a str>,
    #[case] expected: &str,
) {
    let mut cmd = Command::cargo_bin("datafusion-cli").unwrap();
    cmd.args(args);
    cmd.assert().stdout(predicate::eq(expected));
}
