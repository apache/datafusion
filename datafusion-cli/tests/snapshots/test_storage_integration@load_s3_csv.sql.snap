---
source: tests/cli_integration.rs
info:
  program: datafusion-cli
  args: []
  stdin: "CREATE EXTERNAL TABLE CARS\nSTORED AS CSV\nOPTIONS(\n    'aws.access_key_id' 'DataFusionLogin',\n    'aws.secret_access_key' 'DataFusionPassword',\n    'aws.endpoint' 'http://localhost:9000',\n    'aws.allow_http' 'true'\n)\nLOCATION 's3://datafusion/cars.csv';\n\nSELECT * FROM CARS limit 1;"
input_file: tests/sql/load_s3_csv.sql
---
success: true
exit_code: 0
----- stdout -----
[CLI_VERSION]
0 row(s) fetched. 
[ELAPSED]

+-----+-------+---------------------+
| car | speed | time                |
+-----+-------+---------------------+
| red | 20.0  | 1996-04-12T12:05:03 |
+-----+-------+---------------------+
1 row(s) fetched. 
[ELAPSED]

\q

----- stderr -----
